lib/vtls/openssl.c

```diff
@@ -2356,9 +2356,25 @@ static CURLcode get_cert_chain(struct connectdata *conn,
     if(!pubkey)
       infof(data, "   Unable to load public key\n");
     else {
-      switch(pubkey->type) {
+      int pktype;
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L && \
+    !defined(LIBRESSL_VERSION_NUMBER)
+      pktype = EVP_PKEY_id(pubkey);
+#else
+      pktype = pubkey->type;
+#endif
+      switch(pktype) {
       case EVP_PKEY_RSA:
-        BIO_printf(mem, "%d", BN_num_bits(pubkey->pkey.rsa->n));
+      {
+        RSA *rsa;
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L && \
+    !defined(LIBRESSL_VERSION_NUMBER)
+        rsa = EVP_PKEY_get0_RSA(pubkey);
+        BN_print(mem, rsa->n);
```

Comment:
Aren't macros supposed to be in upper case? So IMHO:
```
print_pubkey_BN -> PRINT_PUBKEY_BN
```

Also remember OpenSSL could have a `#define OPENSSL_NO_RSA` in it's `opensslconf.h`. Hence num-bits should be `0`, or?